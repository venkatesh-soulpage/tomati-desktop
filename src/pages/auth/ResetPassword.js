import React from "react";
//redux
import { connect } from "react-redux";
//router
import { withRouter } from "react-router-dom";
//auth
import {
  forgotPasswordToggle,
  receiveResetPassword,
  receiveForgotPasswordToken,
} from "_actions/auth";
// local component
import ForgotPasswordForm from "forms/ForgotPasswordForm";
import ResetPasswordForm from "forms/ResetPasswordForm";
import { useLocation } from "react-router-dom";

function ResetPassword(props) {
  const [values, setValues] = React.useState({
    email: "",
    token: "",
    password: "",
  });

  const useQuery = () => {
    return new URLSearchParams(useLocation().search);
  };
  const query = useQuery();
  const email = query.get("email");
  const token = query.get("token");
  if (!email && !token) {
  }
  return <ResetPasswordForm />;
}

function mapStateToProps(state) {
  return { auth: state.auth };
}

export default withRouter(connect(mapStateToProps)(ResetPassword));
