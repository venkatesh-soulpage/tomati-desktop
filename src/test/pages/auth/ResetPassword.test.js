import React from "react";
import { render } from "@testing-library/react";
import ResetPassword from "pages/auth/ResetPassword";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("ResetPassword page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ResetPassword />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
