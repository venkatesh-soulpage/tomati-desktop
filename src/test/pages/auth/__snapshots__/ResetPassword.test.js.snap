// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-height mt-5 pt-5"
  >
    <div
      class="container"
    >
      <div
        class="row h-100 justify-content-center mt-5"
      >
        <div
          class="col-md-6 w-100 align-self-center"
        >
          <h2
            class="text-dark mb-0 text-center mb-4"
          >
            Update Password
          </h2>
          <div
            class="p-5 pb-3 card"
            style="border-radius: 15px;"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="New Password"
                    type="password"
                    value=""
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      style="display: flex; justify-content: center; align-items: center; width: 60px; border: 1px solid #ced4da; border-top-right-radius: 5px; border-bottom-right-radius: 5px; background-color: transparent; cursor: pointer;"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <input
                    autocomplete="off"
                    class="form-control border-right-0"
                    name="password"
                    placeholder="Confirm New Password"
                    type="password"
                    value=""
                  />
                  <div
                    class="input-group-append"
                  >
                    <div
                      style="display: flex; justify-content: center; align-items: center; width: 60px; border: 1px solid #ced4da; border-top-right-radius: 5px; border-bottom-right-radius: 5px; background-color: transparent; cursor: pointer;"
                    >
                      <small>
                        Show
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-right"
              >
                <div
                  class="mt-3 form-group"
                >
                  <button
                    class="rounded-pill btn btn-danger"
                    type="button"
                  >
                    Update
                  </button>
                   
                  <a
                    href="/"
                  >
                    <button
                      class="rounded-pill btn btn-light"
                      type="button"
                    >
                      Cancel
                    </button>
                  </a>
                </div>
              </div>
              <div
                class="text-muted"
              >
                <a
                  class="red-link"
                  href="/"
                >
                  Login as a different user
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
